docs/platforms/dotnet/common/performance/instrumentation/custom-instrumentation/caches-module.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: Instrument Caches
+sidebar_order: 1000
+description: "Learn how to manually instrument your code to use Sentry's Caches module. "
+---
+
+A cache can be used to speed up data retrieval, thereby improving application performance. Because instead of getting data from a potentially slow data layer, your application will be getting data from memory (in a best case scenario). Caching can speed up read-heavy workloads for applications like Q&A portals, gaming, media sharing, and social networking.
+
+Sentry offers a [cache-monitoring dashboard](https://sentry.io/orgredirect/organizations/:orgslug/insights/caches/) for getting an overview of your application's caches.
+
+To make it possible for Sentry to give you an overview of your cache performance, you'll need to create two spans - one indicating that something is being put into the cache, and a second one indicating that something is being fetched from the cache.
+
+Make sure that there's a transaction running when you create the spans. If you're using a web framework like Rails those transactions will be created for you automatically. See <PlatformLink to='/performance/'>Performance Monitoring</PlatformLink> for more information.
+
+For detailed information about which data can be set, see the [Cache Module Developer Specification](https://develop.sentry.dev/sdk/performance/modules/caches/).
+
+## Manual Instrumentation
+
+Follow the steps below to make sure your Cache related spans end up in Sentry correctly.
+
+### Add Span When Putting Data Into the Cache
+
+1. Set the cache value with whatever cache library you happen to be using.
+2. Wrap the part of your application that uses the cached value with  `Sentry.with_child_span { |span| ... }`
+3. Set `op` to `cache.put`.
+4. Set `cache.item_size` to an integer representing the size of the cached item.
+
+### Add Span When Retrieving Data From the Cache
+
+1. Fetch the cached value from whatever cache library you happen to be using.
+2. Wrap the part of your application that uses the cached value with  `Sentry.with_child_span { |span| ... }`
+3. Set `op` to `cache.get`.
+4. Set `cache.hit` to a boolean value representing whether the value was successfully fetched from the cache or not.
+5. Set `cache.item_size` to an integer representing the size of the cached item.
+
+### Example
+
+Consider the following example service using `IMemoryCache` allowing you to `SetInCache` and `GetFromCache`.
+
+```csharp
+using Microsoft.Extensions.Caching.Memory;
+
+public class MyCachingService
+{
+    private readonly IMemoryCache _cache;
+
+    public MyCachingService(IMemoryCache cache)
+    {
+        _cache = cache;
+    }
+
+    public void SetInCache(string cacheKey, object value)
+    {
+        var cacheSpan = SentrySdk.GetSpan()?.StartChild("cache.put");
+
+        // Describe the cache server you are accessing
+        cacheSpan?.SetExtra("network.peer.address", "cache.example.com/supercache");
```

Comment:
this will end up in `span.data`? (or does .net does not have this?)

Reply:
Yes. Basically tech-debt that we call it `Extra`.